Libraries/LibGUI/ColorPicker.cpp

```diff
@@ -28,88 +28,264 @@
+    tab_palette->layout()->set_margins({ 4, 4, 4, 4 });
+    tab_palette->layout()->set_spacing(4);
+
+    build_ui_palette(*tab_palette);
+
+    auto tab_custom_color = tab_widget.add_tab<Widget>("Custom Color");
+    tab_custom_color->set_size_policy(SizePolicy::Fill, SizePolicy::Fill);
+    tab_custom_color->set_layout<VerticalBoxLayout>();
+    tab_custom_color->layout()->set_margins({ 4, 4, 4, 4 });
+    tab_custom_color->layout()->set_spacing(4);
+
+    build_ui_custom(*tab_custom_color);
+
+    auto& button_container = root_container.add<Widget>();
+    button_container.set_size_policy(SizePolicy::Fill, SizePolicy::Fixed);
+    button_container.set_preferred_size(0, 22);
+    button_container.set_layout<HorizontalBoxLayout>();
+    button_container.layout()->set_spacing(4);
+    button_container.layout()->add_spacer();
+
+    auto& cancel_button = button_container.add<Button>();
+    cancel_button.set_size_policy(SizePolicy::Fixed, SizePolicy::Fill);
+    cancel_button.set_preferred_size(80, 0);
+    cancel_button.set_text("Cancel");
+    cancel_button.on_click = [this] {
+        done(ExecCancel);
+    };
+
+    auto& ok_button = button_container.add<Button>();
+    ok_button.set_size_policy(SizePolicy::Fixed, SizePolicy::Fill);
+    ok_button.set_preferred_size(80, 0);
+    ok_button.set_text("Select");
+    ok_button.on_click = [this] {
+        done(ExecOK);
+    };
+}
+
+void ColorPicker::build_ui_palette(Widget& root_container)
+{
+    unsigned colors[4][9] = {
+        { 0xef2929, 0xf0b143, 0xfce94f, 0x9fe13a, 0x7c9ece, 0xa680a8, 0xe1ba70, 0x888a85, 0xeeeeec },
+        { 0xba1e09, 0xf57900, 0xe9d51a, 0x8bd121, 0x4164a3, 0x6f517b, 0xb77f19, 0x555753, 0xd4d7cf },
+        { 0x961605, 0xbf600c, 0xe9d51a, 0x619910, 0x2b4986, 0x573666, 0x875b09, 0x2f3436, 0xbbbdb6 },
+        { 0x000000, 0x2f3436, 0x555753, 0x808080, 0xbabdb6, 0xd3d7cf, 0xeeeeec, 0xf3f3f3, 0xffffff }
+    };
 
-    auto& left_vertical_container = horizontal_container.add<Widget>();
-    left_vertical_container.set_layout<VerticalBoxLayout>();
+    for (int r = 0; r < 4; r++) {
+        auto& colors_row = root_container.add<Widget>();
+        colors_row.set_layout<HorizontalBoxLayout>();
+        colors_row.set_size_policy(SizePolicy::Fill, SizePolicy::Fill);
 
-    auto& right_vertical_container = horizontal_container.add<Widget>();
-    right_vertical_container.set_layout<VerticalBoxLayout>();
+        for (int i = 0; i < 8; i++) {
+            create_color_button(colors_row, colors[r][i]);
+        }
+    }
+}
 
+void ColorPicker::build_ui_custom(Widget& root_container)
+{
     enum RGBComponent {
         Red,
         Green,
         Blue
     };
 
-    m_preview_widget = right_vertical_container.add<Frame>();
+    auto& horizontal_container = root_container.add<Widget>();
+    horizontal_container.set_fill_with_background_color(true);
+    horizontal_container.set_layout<HorizontalBoxLayout>();
+    //horizontal_container.layout()->set_margins({ 4, 4, 4, 4 });
```

Comment:
Please don't leave commented-out code. Either put it behind an `#ifdef SOMETHING` or remove it :)